pkg/catalog/routes.go

```diff
@@ -255,48 +254,45 @@ func getTrafficPolicyPerRoute(mc *MeshCatalog, routePolicies map[trafficpolicy.T
 				log.Error().Msgf("TrafficTarget %s/%s could not get source services for service account %s", trafficTargets.Namespace, trafficTargets.Name, fmt.Sprintf("%s/%s", trafficSources.Namespace, trafficSources.Name))
 				return nil, srcErr
 			}
-			srcService := srcServiceList[0]
 
-			trafficTarget := trafficpolicy.TrafficTarget{
-				Name:        trafficTargets.Name,
-				Destination: destService,
-				Source:      srcService,
-			}
+			trafficTargetPermutations := listTrafficTargetPermutations(trafficTargets.Name, srcServiceList, destServiceList)
 
-			for _, trafficTargetSpecs := range trafficTargets.Spec.Rules {
-				if trafficTargetSpecs.Kind != HTTPTraffic {
-					log.Error().Msgf("TrafficTarget %s/%s has Spec Kind %s which isn't supported for now; Skipping...", trafficTargets.Namespace, trafficTargets.Name, trafficTargetSpecs.Kind)
-					continue
-				}
+			for _, trafficTarget := range trafficTargetPermutations {
+				for _, trafficTargetSpecs := range trafficTargets.Spec.Rules {
+					if trafficTargetSpecs.Kind != HTTPTraffic {
+						log.Error().Msgf("TrafficTarget %s/%s has Spec Kind %s which isn't supported for now; Skipping...", trafficTargets.Namespace, trafficTargets.Name, trafficTargetSpecs.Kind)
+						continue
+					}
 
-				specKey := mc.getTrafficSpecName(trafficTargetSpecs.Kind, trafficTargets.Namespace, trafficTargetSpecs.Name)
-				routePoliciesMatched, matchFound := routePolicies[specKey]
-				if !matchFound {
-					log.Error().Msgf("TrafficTarget %s/%s could not find a TrafficSpec %s", trafficTargets.Namespace, trafficTargets.Name, specKey)
-					return nil, errNoTrafficSpecFoundForTrafficPolicy
+					specKey := mc.getTrafficSpecName(trafficTargetSpecs.Kind, trafficTargets.Namespace, trafficTargetSpecs.Name)
```

Comment:
I see an opportunity in this section here to create a few smaller functions so it is easier to see all the `for` loops and with proper function naming should be easy to understand what happens -- but we can do that in  another PR.

Reply:
+1 for doing it in another PR. I think it'll make it easier to understand that way